arm-datalake-store/account/2016-11-01/swagger/account.json

```diff
@@ -0,0 +1,1311 @@
+            "$ref": "#/definitions/FirewallRule"
+          },
+          "description": "the results of the list operation"
+        },
+        "nextLink": {
+          "type": "string",
+          "readOnly": true,
+          "description": "the link (url) to the next page of results."
+        },
+        "count": {
+          "type": "integer",
+          "readOnly": true,
+          "format": "int64",
+          "description": "the total count of results that are available, but might not be returned in the current page."
+        }
+      },
+      "description": "Data Lake Store firewall rule list information."
+    },
+    "EncryptionIdentity": {
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "The type of encryption being used. Currently the only supported type is 'SystemAssigned'.",
+          "enum": [
+            "SystemAssigned"
+          ],
+          "x-ms-enum": {
+            "name": "EncryptionIdentityType",
+            "modelAsString": false
+          }
+        },
+        "principalId": {
+          "type": "string",
+          "format": "uuid",
+          "readOnly": true,
+          "description": "The principal identifier associated with the encryption."
+        },
+        "tenantId": {
+          "type": "string",
+          "format": "uuid",
+          "readOnly": true,
+          "description": "The tenant identifier associated with the encryption."
+        }
+      },
+      "description": "The encryption identity properties."
+    },
+    "EncryptionConfig": {
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "The type of encryption configuration being used. Currently the only supported types are 'UserManaged' and 'ServiceManaged'.",
+          "enum": [
+            "UserManaged",
+            "ServiceManaged"
+          ],
+          "x-ms-enum": {
+            "name": "EncryptionConfigType",
+            "modelAsString": false
+          }
+        },
+        "keyVaultMetaInfo": {
+          "$ref": "#/definitions/KeyVaultMetaInfo",
+          "description": "The Key vault information for connecting to user managed encryption keys."
+        }
+      },
+      "description": "The encryption configuration for the account."
+    },
+    "KeyVaultMetaInfo": {
+      "properties": {
+        "keyVaultResourceId": {
+          "type": "string",
+          "description": "The resource identifier for the user managed Key Vault being used to encrypt."
+        },
+        "encryptionKeyName": {
+          "type": "string",
+          "description": "The name of the user managed encryption key."
+        },
+        "encryptionKeyVersion": {
+          "type": "string",
+          "description": "The version of the user managed encryption key."
+        }
+      },
+      "description": "Metadata information used by account encryption."
+    },
+    "DataLakeStoreAccountProperties": {
+      "properties": {
```

Comment:
Nope everything is optional. I did add requirements in the nested properties, though, if they are passed.